### ISSUE TITLE
WEEK 7 ASSIGNMENT

### ISSUE THREAD

**Issue by IDRIS-MUSA** [IDRIS-MUSA/my-project1#7]
TASK 1: LOAD AND EXPLORE THE DATASET:

import pandas as pd
import numpy as np
from sklearn.datasets import load_iris
import matplotlib.pyplot as plt
import seaborn as sns

Load Iris dataset
iris = load_iris()
df = pd.DataFrame(data=iris.data, columns=iris.feature_names)
df['species'] = iris.target

Map species to names
species_map = {0: 'setosa', 1: 'versicolor', 2: 'virginica'}
df['species'] = df['species'].map(species_map)

Display first few rows
print(df.head())

Check data types and missing values
print(df.info())
print(df.isnull().sum())

TASK 2: BASIC DATA ANALYSIS:

Compute basic statistics
print(df.describe())

Group by species and compute mean
grouped_df = df.groupby('species')[['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']].mean()
print(grouped_df)

TASK 3: DATA VISUALIZATION:

Line chart (not applicable for Iris dataset, using bar chart instead)
plt.figure(figsize=(8, 6))
sns.barplot(x='species', y='sepal length (cm)', data=df)
plt.title('Sepal Length by Species')
plt.xlabel('Species')
plt.ylabel('Sepal Length (cm)')
plt.show()

Bar chart
plt.figure(figsize=(8, 6))
sns.barplot(x='species', y='petal length (cm)', data=df)
plt.title('Petal Length by Species')
plt.xlabel('Species')
plt.ylabel('Petal Length (cm)')
plt.show()

Histogram
plt.figure(figsize=(8, 6))
sns.histplot(df['sepal length (cm)'], kde=True)
plt.title('Distribution of Sepal Length')
plt.xlabel('Sepal Length (cm)')
plt.ylabel('Frequency')
plt.show()

Scatter plot
plt.figure(figsize=(8, 6))
sns.scatterplot(x='sepal length (cm)', y='petal length (cm)', hue='species', data=df)
plt.title('Relationship between Sepal and Petal Length')
plt.xlabel('Sepal Length (cm)')
plt.ylabel('Petal Length (cm)')
plt.legend(title='Species')
plt.show()

NOTE:
The code loads the Iris dataset, explores its structure, performs basic data analysis, and also creates four different types of visualizations.
The plots are customized with titles, labels, and legends to provide insights into the dataset.
